### PR TITLE
Prefer plain unzip for extracting archives instead of PowerShell when available

### PR DESCRIPTION
**Pull request by josteink**
Currently on Linux, Powershell is used to unzip ZIP-archives even when plain unzip is available.

However using PowerShell has more dependencies (versions, needs correct parameters, escaping etc) than plain `unzip`. On Ubuntu powershell is also provided as a snap which may or may not cause issues wrt to environment-variables, permissions, etc.

This PR changes the probing order in `lsp-mode.el` to prefer unzip over PowerShell when unzip is available.

This PR partially adresses: https://github.com/emacs-lsp/lsp-mode/issues/3802

### CODE REVIEW

**Review by kiennq — APPROVED** [review]

**Review by jcs090218 — APPROVED** [review]
CI should be good for now.

**Comment by jcs090218** [comment]
> So if it's still red, it means I broke something?

No, the ert tests seems to be broken in two snapshot (windows and linux) and there is nothing to do with this PR. This PR should be good to merge, but we will have to wait for other maintainers with admin privileges to merge this.

**Comment by yyoncho** [comment]
I believe that we switched them for exactly the same problem but in the opposite direction. But lets see how it goes. We may change the code to be os specific and prefer  unzip for linux and powershell for windows.

**Comment by josteink** [comment]
Makes sense. I was considering how it was "clever" to prove for tools rather than platforms, but maybe in practice that works less great than one would think 🤷

**Pull request merged**
